### PR TITLE
Add possibility to override 'protected' item in Child container

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/40324 and https://github.com/joomla/joomla-cms/pull/40388

### Summary of Changes

Allow to overide resource if it not exist locally

### Testing Instructions

Unittest passes

### Documentation Changes Required

New feature

### CODE REVIEW

File: src/Container.php [hunk 1]
```diff
@@ -207,6 +207,22 @@ public function isProtected(string $resourceName): bool
...
+	/**
+	 * Check whether a resource is stored locally
+	 *
+	 * @param   string  $resourceName  Name of the resource to check.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function isLocal(string $resourceName): bool
```
Comment (laoneo) [inline src/Container.php:219]: As you made this function public, can you add some unit tests for it?

**Comment by laoneo** [comment]
@Fedik can you rebase this one to 3.0-dev as it is a breaking change?

**Comment by laoneo** [comment]
Nice tests, can you rebase to 3?

**Comment by Fedik** [comment]
Let it to decide by repository maintainers. It can be done at any time.

**Comment by laoneo** [comment]
We discussed it in maintainers chat already and it is clear that it is a bc break so it has to go into 3.

**Comment by Fedik** [comment]
Changed, I will look in conflicts later

**Pull request closed without merging**